### DISCUSSION TITLE
Change heating groups programmatically

### DISCUSSION THREAD

**Discussion by simllll** [SukramJ/aiohomematic#1858] [Q&A] [answered]
Hi all,
thanks for all the awesome work, I'm new to the homematic environemnt and still trying to find some solutions to newly learned limitatinos withing homematic ;-).
My issue is that I have a cooling and a heating system (ceiling vs floor). I try to avoid installing additional thermostats for the cooling system, but as homematic / rasperrymatic doesn't allow to assign one thermostat to two "heating groups", I'm looking for a way when I switch e.g. to "summer mode", that the thermostats which are associated to the heating system during winter, will be re-associated to the cooling system during summer.

Is there any possibility to add a way to programtically change (heating) groups? Basically add and remove devices. I checked the current functions, and couldn't ifnd a way.. I also couldn't find a way direclty in the CCU, but it seems the NEOServer can handle such things. And if I get it right, the communication from HA to CCU is done via NEOServer? Even if not, my question remains ;-) 

Or anyone has another idea how I could solve this quest?

Thanks a lot
Simon

**Comment by SukramJ** [comment, marked as answer]
> Is there any possibility to add a way to programtically change (heating) groups?

Nein. 

>  And if I get it right, the communication from HA to CCU is done via NEOServer?

Nein

**Reply by simllll** [reply]
Danke für die schnelle Antwort, meinst du mit "Nein" dass es zurzeit nicht möglich ist, oder dass RasperryMatic aktuell keine interne/API bietet das umzusetzen? Wie wird aktuell mit der CCU kommuniziert? Sind das "offiziele" APIs die die CCU von haus aus mit bringt? Gibt es dazu wo eine Dokumentation welche Endpunkte da unterstützt werden?

**Reply by SukramJ** [reply]
> Danke für die schnelle Antwort, meinst du mit "Nein" dass es zurzeit nicht möglich ist, oder dass RasperryMatic aktuell keine interne/API bietet das umzusetzen?

Beides. Interne APIs kenne ich nicht.

> Wie wird aktuell mit der CCU kommuniziert? 

XML-RPC und JSON-RPC

> Sind das "offiziele" APIs die die CCU von haus aus mit bringt?

Natürlich

> Gibt es dazu wo eine Dokumentation welche Endpunkte da unterstützt werden?

https://homematic-ip.com/sites/default/files/downloads/HM_XmlRpc_API.pdf
https://homematic-ip.com/sites/default/files/downloads/HMIP_XmlRpc_API_Addendum.pdf
https://homematic-ip.com/sites/default/files/downloads/hmip-device-documentation_1.pdf
HomeMatic JSON API:http://"deine CCU IP"/api/homematic.cgi
